staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -326,12 +288,38 @@ func estimateMaxStringLengthPerRequest(s *schema.Structural) int64 {
 	}
 }
 
+// estimateMaxArrayItemsPerRequest estimates the maximum number of array items with
+// the provided minimum serialized size that can fit into a single request.
+func estimateMaxArrayItemsFromMinSize(minSize int64) int64 {
+	// subtract 2 to account for [ and ]
+	return (maxRequestSizeBytes - 2) / (minSize + 1)
+}
+
 // estimateMaxAdditionalPropertiesPerRequest estimates the maximum number of additional properties
-// with the provided schema that can fit into a single request.
-func estimateMaxAdditionalPropertiesPerRequest(additionalPropertiesSchema *schema.Structural) int64 {
+// with the provided minimum serialized size that can fit into a single request.
+func estimateMaxAdditionalPropertiesFromMinSize(minSize int64) int64 {
 	// 2 bytes for key + "" + colon + comma + smallest possible value, realistically the actual keys
 	// will all vary in length
-	keyValuePairSize := estimateMinSizeJSON(additionalPropertiesSchema) + 6
+	keyValuePairSize := minSize + 6
 	// subtract 2 to account for { and }
 	return (maxRequestSizeBytes - 2) / keyValuePairSize
 }
+
+func estimateMinSizeForMap(s *schema.Structural) int64 {
```

Comment:
I'm not sure we need this function, see the comment at the call site. If we do end up needing this function, copy the explanatory comments about required fields, default values, and what the extra 4 is for (quotes, colon, and comma)?

Reply:
Fixed